### PR TITLE
Don't pass doctest feature by default

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
As asked in https://github.com/rust-lang/rust/pull/63803#issuecomment-539064434.

r? @QuietMisdreavus

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
cc @ollie27 @Mark-Simulacrum

File: src/test/rustdoc-ui/doc-test-doctest-feature.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// build-pass
+// compile-flags:--test
+// normalize-stdout-test: "src/test/rustdoc-ui" -> "$$DIR"
+
+#![feature(cfg_doctest)]
+
+// Make sure `cfg(doctest)` is not set when finding doctests but not inside
```
Comment (ollie27) [inline src/test/rustdoc-ui/doc-test-doctest-feature.rs]:
```suggestion
// Make sure `cfg(doctest)` is set when finding doctests but not inside
```

File: src/librustdoc/test.rs [hunk 2]
```diff
@@ -64,6 +64,9 @@ pub fn run(options: Options) -> i32 {
 
     let mut cfgs = options.cfgs.clone();
     cfgs.push("rustdoc".to_owned());
+    if options.should_test {
```
Comment (ollie27) [inline src/librustdoc/test.rs]: You can remove the `options.should_test` check here because it will always be `true` in this function.

**Review by ollie27 — COMMENTED** [review]
r=me with those minor issues fixed.

File: src/librustdoc/test.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply src/librustdoc/test.rs]: Ah indeed, good catch!

**Comment by tmandry** [comment]
@bors retry
have to r+ the rollup first...

**Pull request merged**